### ISSUE TITLE
False positive constructor parameter collides with a type parameter

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#211]
```julia
julia> lintstr("""
       type X{T}
           val::T
           (::Type{X{T}}){T}(n::T) = new{T}(n)
       end
       """)
1-element Array{Lint.LintMessage,1}:
 none:1 E523 T: constructor parameter collides with a type parameter
```
